prov/efa/src/efa_av.c

```diff
@@ -329,7 +329,7 @@ int efa_av_insert_addr(struct efa_av *av, struct efa_ep_addr *addr,
 			" rdm_fiaddr = %" PRIu64 " shm_rdm_fiaddr = %" PRIu64
 			"\n", smr_name, *(uint64_t *)addr, *fi_addr, shm_fiaddr);
 
-		assert(shm_fiaddr < EFA_SHM_MAX_AV_COUNT);
+		assert(shm_fiaddr < rxr_env.shm_av_size);
```

Comment:
Is assertion appropriate here? What will happen when application try to insert more than 128 address in shm?

Reply:
Yes, it's necessary. If apps try to insert more than rxr_env.shm_av_size address in shm, shm still can return a fi_addr, so we need an assertion here to check if the fi_addr exceeds rxr_env.shm_av_size. 

The reason shm still can return a fi_addr is that shm uses ofi_av_* to init/insert/remove. ofi_av_* creates a ofi_buf_pool to store these address. If the buf pool has been used up, it will grow, and return an index(fi_addr) after growing.

Reply:
In this case there should be error handling instead of just assertion. Assertion is only used in debug mode. Before calling ```fi_av_insert```, we should check ```shm_rdm_av``` size, and if it reach ```shm_av_size```, we should  return error, and not calling ```fi_av_insert```.

Reply:
Yeah, it makes sense. I pushed a new revision with both fixes.